addons/account/models/account.py

```diff
@@ -975,6 +976,39 @@ def action_configure_bank_journal(self):
         # We simply call the setup bar function.
         return self.env['res.company'].setting_init_bank_account_action()
 
+    @api.multi
+    def create_invoice_from_attachment(self, attachment_ids=[]):
+        ''' Create the invoices from files.
+         :return: A action redirecting to account.invoice tree/form view.
+        '''
+        attachments = self.env['ir.attachment'].browse(attachment_ids)
+        if not attachments:
+            raise UserError(_("No attachment was provided"))
+
+        invoices = self.env['account.invoice']
+        journal_id = self or self.env['account.journal'].search([('type', '=', self.env.context.get('journal_type'))], limit=1)
```

Comment:
journal_type = self.env.context.get('journal_type', self.journal_type) or 'purchase'
in case we call this method without the key in context and without self, that way we still have a default.

Then you can write
journal_id = self or self.env['account.journal'].search([('type', '=', journal_type)], limit=1)

Also since this method potentially uses self.id you should add an self.ensure_one() if self is not empty at the start